docs/igor0806/GamesProject/scripts/games.js

```diff
@@ -0,0 +1,136 @@
+        const apiUrl = 'https://mmo-games.p.rapidapi.com/games';
+        showLoading();
+
+        const option = {
+            method: 'GET',
+            mode: 'cors',
+            cache: 'no-cache',
+            headers: {
+                'X-RapidAPI-Key': '1c3169c707mshb51bff34cbc9ff6p1749b9jsn648a19134256',
+                'X-RapidAPI-Host': 'mmo-games.p.rapidapi.com',
+            },
+        };
+
+        const response = await fetch(apiUrl, option);
+        games = await response.json();
+        renderCards(gameList, games.slice(0, 50));
+    } catch (err) {
+        console.error(err, 'Get error:');
+    } finally {
+        noneLoading();
+    }
+}
+
+document.addEventListener('DOMContentLoaded', fetchCards);
+
+function filterGames() {
+    const newGamesCheck = document.querySelector('[data-type-filter="new"]');
+    const oldGamesCheck = document.querySelector('[data-type-filter="old"]');
+    const genreSelect = document.querySelector('[data-type-filter="genre"]');
+    const platformSelect = document.querySelector('[data-type-filter="platform"]');
+    const searchInput = document.querySelector('[data-type-filter="search"]');
+
+    const isNewChecked = newGamesCheck.checked;
+    const isOldChecked = oldGamesCheck.checked;
+    const selectedGenre = genreSelect.value;
+    const selectedPlatform = platformSelect.value;
+    const searchResult = searchInput.value.toLowerCase();
+
+    const filteredGames = games.filter((game) => {
+        const isGenreMatch = selectedGenre === 'Genre' || game.genre.includes(selectedGenre);
+        const isPlatformMatch = selectedPlatform === 'Platform' || game.platform.includes(selectedPlatform);
+        const isSearchMatch = game.title.toLowerCase().includes(searchResult);
+
+        return isGenreMatch && isPlatformMatch && isSearchMatch;
+    });
+
+    if (isNewChecked) {
+        filteredGames.sort((a, b) => new Date(b.release_date) - new Date(a.release_date));
+    } else if (isOldChecked) {
+        filteredGames.sort((a, b) => new Date(a.release_date) - new Date(b.release_date));
+    }
+
+    renderCards(gameList, filteredGames);
+}
+
+function init() {
+    const filterForm = document.querySelector('[data-type="filter-form"]');
+    const newCheckbox = document.querySelector('[data-type-filter="new"]');
+    const oldCheckbox = document.querySelector('[data-type-filter="old"]');
+    const genreSelect = document.querySelector('[data-type-filter="genre"]');
+    const platformSelect = document.querySelector('[data-type-filter="platform"]');
+    const searchButton = document.querySelector('[data-type-filter="search-button"]');
+
+    filterForm.addEventListener('change', filterGames);
+    newCheckbox.addEventListener('change', filterGames);
+    oldCheckbox.addEventListener('change', filterGames);
+    genreSelect.addEventListener('change', filterGames);
+    platformSelect.addEventListener('change', filterGames);
+
+    searchButton.addEventListener('click', (event) => {
```

Comment:
Now if you type some text in the search field and hit Enter, all page reloads and all chosen filters are gone